src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs

```diff
@@ -32,5 +33,49 @@ public static string ExtractSoapAction(this HttpRequest request, IApmLogger logg
 
 			return null;
 		}
+
+		/// <summary>
+		/// Extracts the soap action from the header if exists only with Soap 1.1
+		/// </summary>
+		/// <param name="request">The request.</param>
+		/// <param name="logger">The logger.</param>
+		private static string GetSoap11Action(HttpRequest request)
+		{
+			var soapActionWithNamespace = request.Headers.Get(SoapActionHeaderName);
+			if (!string.IsNullOrWhiteSpace(soapActionWithNamespace))
+			{
+				var indexPosition = soapActionWithNamespace.LastIndexOf(@"/", StringComparison.InvariantCulture);
+				if (indexPosition != -1) return soapActionWithNamespace.Substring(indexPosition + 1).TrimEnd('\"');
+			}
+			return null;
+		}
+
+		/// <summary>
+		/// Lightweight parser that extracts the soap action from the xml body only with Soap 1.2
+		/// </summary>
+		/// <param name="request">The request.</param>
+		/// <param name="logger">The logger.</param>
+		private static string GetSoap12Action(HttpRequest request)
+		{
+			//[{"key":"Content-Type","value":"application/soap+xml; charset=utf-8"}]
+			var contentType = request.Headers.Get(ContentTypeHeaderName);
+			if (contentType?.Contains(SoapAction12ContentType) != true) return null;
+			if (!request.InputStream.CanSeek) return null;
+
+			try
+			{
+				var xmlReader = new XmlTextReader(request.InputStream);
+				xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+				xmlReader.Read();
+				xmlReader.ReadStartElement(); // if (xmlReader.LocalName != "Body") return null;
+				xmlReader.ReadStartElement();
+				var action = xmlReader?.LocalName;
```

Comment:
Unfortunately I don't think this logic is enough to parse the Body. 

For example [on this page](https://www.w3.org/TR/soap12-part1/) there is a sample, which this would not parse correctly.

I'd say 1 `Read()` and 2 `ReadStartElement()` is not guaranteed to hit the `Body` part in the xml.